tests/ttnn/unit_tests/gtests/test_broadcast_to.cpp

```diff
@@ -0,0 +1,89 @@
+// SPDX-FileCopyrightText: © 2023 Tenstorrent Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#include <cstdint>
+#include <optional>
+#include "tests/tt_metal/tt_metal/common/dispatch_fixture.hpp"
+#include "ttnn/device.hpp"
+#include "ttnn/operations/experimental/bcast_to/bcast_to.hpp"
+#include "ttnn/operations/core/core.hpp"
+#include "ttnn/operations/creation.hpp"
+#include "ttnn_test_fixtures.hpp"
+
+namespace ttnn {
+namespace operations {
+namespace experimental {
+namespace broadcast_to {
+namespace test {
+
+struct BroadcastParam {
+    uint32_t n;  // input tensor batch
+    uint32_t c;  // input tensor channel
+    uint32_t h;  // input tensor height
+    uint32_t w;  // input tensor width
+    std::vector<int32_t> broadcast_shape;
+};
+
+class Broadcast_toFixture : public TTNNFixtureWithDevice, public testing::WithParamInterface<BroadcastParam> {};
+
+TEST_P(Broadcast_toFixture, Broadcast_to) {
+    auto param = GetParam();
+    auto& device = *device_;
+    std::array<uint32_t, 4> dimensions = {param.n, param.c, param.h, param.w};
+    ttnn::Shape input_shape(dimensions);
+    std::vector<uint32_t> output_size = {
+        param.broadcast_shape[0],
+        param.broadcast_shape[1],
+        param.broadcast_shape[2],
+        param.broadcast_shape[3],
+    };
+    ttnn::Shape output_shape(output_size);
```

Comment:
If you make `BroadcastParam::broadcast_shape` a `std::vector<uint32_t>` then you could just...
```suggestion
    ttnn::Shape output_shape(param.broadcast_shape);
```